lib/memcached/memcached.rb

```diff
@@ -585,18 +599,46 @@ def stats(subcommand = nil)
     raise e
   end
 
+  # Causes get / cas operations to return NOT_FOUND instead of raising NotFound
+  def without_exceptions(&block)
+    prev = @raise_on_errors
+    @raise_on_errors = false
+    yield
+  ensure
+    @raise_on_errors = prev
+  end
+
   ### Operations helpers
 
   private
 
   # Checks the return code from Rlibmemcached against the exception list. Raises the corresponding exception if the return code is not Memcached::Success or Memcached::ActionQueued. Accepts an integer return code and an optional key, for exception messages.
+  # When @raise_on_errors is false the following are returned instead of raiing exceptions:
+  # `NOT_STORED` is returned when ret == 12 (Lib::MEMCACHED_DATA_EXISTS)
+  # `NOT_FOUND` is returned when ret == 16 (Lib::MEMCACHED_NOTFOUND)
+  # `DATA_EXISTS` is returned when ret == 14 (Lib::MEMCACHED_NOTSTORED)
   def check_return_code(ret, key = nil) #:doc:
```

Comment:
I see where @casperisfine is going. This function feels a bit awkward now, if we dropped the exceptions completely (not sure how much work that is) then it'd be called something like  `return_code` and just give back a value instead of the nil check. ( I guess we could say now is a bit golang inspired `if err != nil` )